Source/WebCore/html/HTMLLabelElement.cpp

```diff
@@ -140,6 +143,14 @@ bool HTMLLabelElement::isEventTargetedAtInteractiveDescendants(Event& event) con
 void HTMLLabelElement::defaultEventHandler(Event& event)
 {
     if (event.type() == eventNames().clickEvent && !m_processingClick) {
+        // If text of label element is selected, do not pass the event to control element.
+        // Note: a click event may be not a mouse event if created by document.createEvent().
+        if (event.isMouseEvent() && !downcast<MouseEvent>(event).isSimulated()) {
+            auto* frame = document().frame();
+            if (frame && frame->selection().selection().isRange())
```

Comment:
Can you test the case where the label is already selected, and then you click on the selected text? Would like to make sure the control element is still focused in that case.

Reply:
I found the desired test in Chrome / Blink here - [Link](https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/web_tests/fast/forms/label/label-selection-by-textSelection-and-click.html) but it does not work in WebKit and similarly, I created this pre-selected label text fiddle ([link](https://jsfiddle.net/dao3m021/1/)), when double clicking non-patched system, I am unable to get text selection with double click while after patched local, I am able to get selection. (Single click still moves to Input focus similar to Chrome Canary).

Reply:
@pxlcoder - Anything on above from your side?

Reply:
> Single click still moves to Input focus similar to Chrome Canary

Sounds good!